src/database/subClasses/ExperimentMethods.java

```diff
@@ -72,30 +76,30 @@ public Experiment getExperiment(String expID) throws SQLException {
      * @throws SQLException
      *             if the query does not succeed
      * @throws IOException
-     * @throws DuplicatePrimaryKeyException
      *             If the experiment already exists.
      */
     public int addExperiment(String expID) throws SQLException, IOException {
+        if (expID == null || expID.isEmpty()) {
+            throw new IOException("Invalid experiment ID.");
+        }
 
-    	if (expID == null || expID.isEmpty()) {
-    		throw new IOException("Invalid experiment ID.");
-    	}
-
-    	Experiment e = getExperiment(expID);
-    	if (e != null) {
-    		throw new IOException(expID + " already exists");
-    	}
+        Experiment e = getExperiment(expID);
+        if (e != null) {
+            throw new IOException(expID + " already exists");
+        }
 
+        fpg.generateExperimentFolders(expID);
         String query = "INSERT INTO Experiment "
                 + "(ExpID) VALUES (?)";
-        PreparedStatement stmt = conn.prepareStatement(query);
-        stmt.setString(1, expID);
-
-        int rs = stmt.executeUpdate();
-        stmt.close();
-
-        fpg.generateExperimentFolders(expID);
-
+        int rs = 0;
+        try (PreparedStatement stmt= conn.prepareStatement(query);) {
+            stmt.setString(1, expID);
+            rs = stmt.executeUpdate();
+            stmt.close();
+        } catch (SQLException ex) {
+            conn.rollback();
+            throw ex;
+        }
```

Comment:
That was just a reminder of how things are done otherwise. Ofcourse it's not applicable in this case due to the aforementioned reason.